jax/_src/numpy/lax_numpy.py

```diff
@@ -3847,6 +3847,36 @@ def diagflat(v, k=0):
   res = res.reshape(adj_length,adj_length)
   return res
 
+@_wraps(np.poly)
+def poly(seq_of_zeros):
+  seq_of_zeros = atleast_1d(seq_of_zeros)
+
+  dt = seq_of_zeros.dtype
+  seq_of_zeros = seq_of_zeros.astype(np.mintypecode(dt.char))
```

Comment:
Is this line necessary? In what contexts will it result in the type of the array being changed?

Reply:
Seems like it's probably not something we want; e.g.
```python
>>> dtype = np.dtype(jnp.bfloat16)                                                                                                                           
>>> np.dtype(np.mintypecode(np.dtype(jnp.bfloat16).char))                                                                                                    
dtype('float64')
```

Reply:
In https://github.com/numpy/numpy/issues/5096, there's an issue where `np.poly` doesn't work as expected for integer arrays, so the solution that they implemented to fix it was to turn them into `float64` arrays.

I wrote this implementation to try and be similar to numpy's implementation (even if it might not be the best). I agree that this line is probably not-ideal behavior, so I can change that if it's okay for there to be a tiny difference between jax's behavior and numpy's behavior.

Reply:
I see, that makes sense. In JAX we generally handle that using `_promote_dtypes_inexact`, which will do the promotion to the appropriate float type following JAX's type promotion logic.

Reply:
https://github.com/google/jax/commit/79d331db2fa4275f0cea4c53131c087398b9fb56 uses `_promote_dtypes_inexact`, which removes the need for `np.mintypecode`.